lib/tasks/encrypt.rake

```diff
@@ -24,33 +24,37 @@ end
 end
 
 namespace :db do
+  def modify_encryptable_fields(klass, action)
+    klass.order(:id).each do |auth_source_ldap|
+      auth_source_ldap.encryptable_fields.each do |field|
+        str = auth_source_ldap.read_attribute(field.to_sym)
+        auth_source_ldap.update_column(field.to_sym,
+                                       auth_source_ldap.send("#{action}_field", str))
+      end
+    end
+  end
+
+  namespace :auth_sources_ldap do
+    desc "Encrypt LDAP authentication source fields"
+    task :encrypt => :environment do
+      modify_encryptable_fields(AuthSourceLdap, :encrypt)
+    end
+
+    desc "Decrypt compute resource fields"
+    task :decrypt => :environment do
+      modify_encryptable_fields(AuthSourceLdap, :decrypt)
+    end
+  end
+
   namespace :compute_resources do
```

Comment:
Or, thinking about it again, have the separate rake tasks (which is handy for me testing this) and have one super-task that lists them as dependencies.

The one task might be useful for plugins as they can add their own encryption tasks to it.